sway/desktop/render.c

```diff
@@ -2004,8 +2212,12 @@ void output_render(struct sway_output *output, struct timespec *when,
 			render_output_blur(output, damage);
 		}
 
-		render_workspace(output, damage, workspace, workspace->current.focused);
-		render_floating(output, damage);
+		render_workspace(output, damage, !fullscreen_con ? workspace : NULL,
+				workspace->current.focused,
+				!other_ws_has_fullscreen ? other_ws : NULL);
+		render_floating(output, damage,
+				!other_ws_has_fullscreen ? other_ws : NULL,
+				fullscreen_con != NULL);
```

Comment:
Atm, we only adjust the floating and tiled views while switching workspaces. Other desktops handle this differently. Gnome also translates the background and macOS and Elementary OS translates all the windows/panels.

Should we change the current behavior?

Reply:
I think the default is good, changing between wallpapers and panels looks goofy to me (although I know the whole point of SwayFX is user customization), let's get it in as is for now and we can always add those options later

Reply:
> I think the default is good, changing between wallpapers and panels looks goofy to me (although I know the whole point of SwayFX is user customization), let's get it in as is for now and we can always add those options later

Sounds good to me! I was experimenting with it yesterday, but I somehow broke the damage tracking xD

It was a bit more involved than the current implementation, so I'm fine with sticking with how it is now :)